packages/cli/src/cmds/preview/previewEnv.ts

```diff
@@ -3,71 +3,217 @@
+    let result = true;
+    let summaryMsg = `${Progress.M365Account}`;
+    let error = undefined;
+    const accountBar = CLIUIInstance.createProgressBar(Progress.M365Account, 1);
+    await accountBar.start(ProgressMessage[Progress.M365Account]);
+    await accountBar.next(ProgressMessage[Progress.M365Account]);
+    let loginHint: string | undefined = undefined;
+    let tenantId: string | undefined = undefined;
+    try {
+      let loginStatusRes = await M365TokenInstance.getStatus({ scopes: AppStudioScopes });
+      let token = loginStatusRes.isOk() ? loginStatusRes.value.token : undefined;
+      if (loginStatusRes.isOk() && loginStatusRes.value.status === signedOut) {
+        const tokenRes = await M365TokenInstance.getAccessToken({
+          scopes: AppStudioScopes,
+          showDialog: true,
+        });
+        token = tokenRes.isOk() ? tokenRes.value : undefined;
+        loginStatusRes = await M365TokenInstance.getStatus({ scopes: AppStudioScopes });
+      }
+      if (token === undefined) {
+        result = false;
+        summaryMsg = constants.doctorResult.NotSignIn;
+      } else {
+        const isSideloadingEnabled = await getSideloadingStatus(token);
+        if (isSideloadingEnabled === false) {
+          // sideloading disabled
+          result = false;
+          summaryMsg = constants.doctorResult.SideLoadingDisabled;
+        }
+      }
+      const tokenObject = loginStatusRes.isOk() ? loginStatusRes.value.accountInfo : undefined;
+      if (tokenObject && tokenObject.upn) {
+        loginHint = tokenObject.upn as string;
+      }
+      if (tokenObject && tokenObject.tid) {
+        tenantId = tokenObject.tid as string;
+      }
+    } catch (err: any) {
+      result = false;
+      error = assembleError(err, cliSource);
+    }
+    if (result && loginHint) {
+      summaryMsg = constants.doctorResult.SignInSuccess.split("@account").join(`${loginHint}`);
+    }
+    await accountBar.next(summaryMsg);
+    await accountBar.end(result);
+    if (!result) {
+      return error ? err(error) : err(errors.PrerequisitesValidationM365AccountError(summaryMsg));
+    }
+    if (
+      tenantId !== undefined &&
+      appTenantId !== undefined &&
+      tenantId.toLowerCase() !== appTenantId.toLowerCase()
+    ) {
+      cliLogger.necessaryLog(LogLevel.Warning, constants.m365SwitchedMessage);
+    }
+    return ok({ tenantId: tenantId, loginHint: loginHint });
+  }
+
+  protected async detectRunCommand(projectPath: string): Promise<
+    Result<
+      {
+        runCommand: string;
+      },
+      FxError
+    >
+  > {
+    let runCommand: string | undefined = undefined;
+    const hasPackageJson = await fs.pathExists(path.join(projectPath, "package.json"));
+    const csprojs = (await fs.readdir(projectPath)).filter(
+      (f) => path.extname(f).toLowerCase() === ".csproj"
+    );
+    const hasCsproj = csprojs.length === 1;
+    if (hasPackageJson && !hasCsproj) {
+      // package.json should have "dev:teamsfx"
```

Comment:
So we need to update all templates/samples to use this script.

Reply:
Yes